samples/apps/copilot-chat-app/webapi/Controllers/DocumentImportController.cs

```diff
@@ -181,11 +181,21 @@ private async Task ParseDocumentContentToMemoryAsync(IKernel kernel, string cont
 
         foreach (var paragraph in paragraphs)
         {
-            await kernel.Memory.SaveInformationAsync(
+            var memories = kernel.Memory.SearchAsync(
                 collection: targetCollectionName,
-                text: paragraph,
-                id: Guid.NewGuid().ToString(),
-                description: $"Document: {documentName}");
+                query: paragraph,
+                limit: 1,
+                minRelevanceScore: this._options.DeduplicationSimilarityThreshold
+            ).ToEnumerable();
+
+            if (!memories.Any())
```

Comment:
Don't you need to away before using memories?

Is SearchAsync returning an IAsyncEnumerable?

Reply:
`ToEnumerable` is applied on the IAsyncEnumerable instance.